nessie/src/main/java/org/apache/iceberg/nessie/NessieTableOperations.java

```diff
@@ -79,18 +79,19 @@ protected String tableName() {
   }
 
   @Override
-  protected void refreshFromMetadataLocation(String newLocation, Predicate<Exception> shouldRetry,
-      int numRetries) {
+  protected void refreshFromMetadataLocation(String newLocation, Predicate<Exception> shouldRetry, int numRetries) {
     super.refreshFromMetadataLocation(newLocation, shouldRetry, numRetries, this::loadTableMetadata);
   }
 
   private TableMetadata loadTableMetadata(String metadataLocation) {
     // Update the TableMetadata with the Content of NessieTableState.
-    return TableMetadataParser.read(io(), metadataLocation)
-        .withCurrentSnapshotOnly(table.getSnapshotId())
-        .withCurrentSchema(table.getSchemaId())
-        .withDefaultSortOrder(table.getSortOrderId())
-        .withDefaultSpec(table.getSpecId());
+    return TableMetadata.buildFrom(TableMetadataParser.read(io(), metadataLocation))
```

Comment:
@rymurr, I updated this to use the new builder instead of adding mutations to `TableMetadata`.

I think this is a pretty good validation of the builder API -- I was able to use it for this update without adding new operations.